segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerMetadata.java

```diff
@@ -155,21 +159,25 @@ public int getActiveSegmentCount() {
     @Override
     public UpdateableSegmentMetadata mapStreamSegmentId(String streamSegmentName, long streamSegmentId) {
         StreamSegmentMetadata segmentMetadata;
+        int count;
         synchronized (this.lock) {
             validateNewMapping(streamSegmentName, streamSegmentId);
             segmentMetadata = new StreamSegmentMetadata(streamSegmentName, streamSegmentId, getContainerId());
             this.metadataByName.put(streamSegmentName, segmentMetadata);
             this.metadataById.put(streamSegmentId, segmentMetadata);
+            count = this.metadataById.size();
         }
 
         segmentMetadata.setLastUsed(getOperationSequenceNumber());
-        log.info("{}: MapStreamSegment SegmentId = {}, Name = '{}'", this.traceObjectId, streamSegmentId, streamSegmentName);
+        log.info("{}: MapStreamSegment SegmentId = {}, Name = '{}', Active = {}", this.traceObjectId, streamSegmentId, streamSegmentName, count);
+        this.metrics.segmentCount(count);
```

Comment:
What is this counting exactly? The number of segments cached in memory?

Reply:
Exactly. It counts (sets the value of) the active segments per container.